pandas/tests/reshape/test_pivot.py

```diff
@@ -142,6 +142,30 @@ def test_pivot_table_dropna_categoricals(self, dropna):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_pivot_with_non_observable_dropna_civ(self, dropna):
+        # gh-21370
+        arr = [np.nan, 'low', 'high', 'low', np.nan]
+        df = pd.DataFrame({
+            "In": pd.Categorical(arr,
+                                 categories=['low', 'high'],
+                                 ordered=True),
+            "Col": ["A", "B", "C", "A", "B"],
```

Comment:
This should also be a `Categorical` with `NaN` values present for both the `index`/`columns`/`values` case, and the `columns`/`values` case in order to recreate the issue.